pkg/solana/logpoller/job_get_block.go

```diff
@@ -0,0 +1,144 @@
+package logpoller
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/gagliardetto/solana-go"
+	"github.com/gagliardetto/solana-go/rpc"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/client"
+)
+
+// getBlockJob is a job that fetches blocks with transactions, coverts logs into ProgramEvents and writes them into blocks channel
+type getBlockJob struct {
+	slotNumber       uint64
+	client           RPCClient
+	blocks           chan Block
+	done             chan struct{}
+	parseProgramLogs func(logs []string) []ProgramOutput
+	lggr             logger.SugaredLogger
+}
+
+func newGetBlockJob(client RPCClient, blocks chan Block, lggr logger.SugaredLogger, slotNumber uint64) *getBlockJob {
+	return &getBlockJob{
+		client:           client,
+		blocks:           blocks,
+		slotNumber:       slotNumber,
+		done:             make(chan struct{}),
+		parseProgramLogs: parseProgramLogs,
+		lggr:             lggr,
+	}
+}
+
+func (j *getBlockJob) String() string {
+	return fmt.Sprintf("getBlock for slotNumber: %d", j.slotNumber)
+}
+
+func (j *getBlockJob) Done() <-chan struct{} {
+	return j.done
+}
+
+func (j *getBlockJob) Run(ctx context.Context) error {
+	var excludeRewards bool
+	version := client.MaxSupportTransactionVersion
+	block, err := j.client.GetBlockWithOpts(
+		ctx,
+		j.slotNumber,
+		// NOTE: any change to the filtering arguments may affect calculation of txIndex, which could lead to events duplication.
```

Comment:
Are the filtering arguments this is referring to `MaxSupportedTransactionVersion` and `Rewards`?

I wonder if `MaxSupportedTransactionVersion` might change 🤔 if so, that does seem like something we should be very careful about handling--no action requested just thinking we might ask someone about this or note it somewhere to keep track.

Reply:
The version change should be okay. If a requested block contains a transaction with a version greater than `MaxSupportedTransactionVersion,` we'll get an error.
At the moment, only `Rewards` filters transactions, so any change to it will affect transaction's index => affect log's index